### PR TITLE
Feat/gateway docker file

### PR DESCRIPTION
**Pull request by Nullably**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
- create a docker file for Gateway v2
- add hook for auto build to docker hub 

**Tests performed by the developer**:
- docker image can be created from docker file
- docker image is created on docker hub

**Tips for QA testing**:
- docker image and container created from this docker file cannot be used yet as it's still required further process to update environment variables and create config file

### CODE REVIEW

File: gateway/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+WORKDIR /usr/src/app
+
+# install dependancies
+COPY package*.json ./
+# COPY yarn.lock ./
+
+RUN yarn
+
+# copy pwd file to container
+COPY . .
```
Comment (vic-en) [inline gateway/Dockerfile:24]:
```suggestion
# copy pwd file to container
COPY . .

# install dependencies
RUN yarn
```

Comment (vic-en) [reply gateway/Dockerfile:24]: What do you this about this approach of copying files first then going on with the rest?

**Review by vic-en — CHANGES_REQUESTED** [review]
Also I believe `.gitignore` file should be updated to ignore some sensitive gateway files.

File: gateway/Dockerfile (hunk 1 above)
Comment (Nullably) [reply gateway/Dockerfile:24]: I am fine with that.

**Comment by Nullably** [comment]
> Also I believe `.gitignore` file should be updated to ignore some sensitive gateway files.

Which files are you referring to?

**Comment by vic-en** [comment]
> > Also I believe `.gitignore` file should be updated to ignore some sensitive gateway files.
> 
> Which files are you referring to?

Nvm, I wanted to mention `.gitignore`. But I see it doesn't need at update after checking again.

**Review by vic-en — APPROVED** [review]
LGTM

**Pull request merged**